tests/Test/Dscp/Educator/Validate.hs

```diff
@@ -0,0 +1,96 @@
+module Test.Dscp.Educator.Validate where
+
+import Test.Common
+
+import Dscp.Core (ATGDelta (..), CourseId (..), Grade (..))
+import Dscp.Crypto (PublicKey, SecretKey, fromFoldable, getMerkleRoot, hash, sign)
+import Dscp.Educator (PrivateBlock (..), PrivateBlockBody (..), PrivateBlockHeader (..),
+                      PrivateTxAux (..), PrivateTxWitness (..), genesisHeaderHash, validate)
+
+import qualified Data.Map.Strict as M
+
+courseCompScience1 :: CourseId
+courseCompScience1 = CourseId 3
+
+studentAPubKey, studentBPubKey,
+  educatorJPubKey, educatorKPubKey :: PublicKey
+studentAPubKey = mkPubKey 'a'
+studentBPubKey = mkPubKey 'b'
+educatorJPubKey = mkPubKey 'j'
+educatorKPubKey = mkPubKey 'k'
+
+studentAPrivKey, studentBPrivKey,
+  educatorJPrivKey, educatorKPrivKey :: SecretKey
+studentAPrivKey = mkPrivKey 'a'
+studentBPrivKey = mkPrivKey 'b'
+educatorJPrivKey = mkPrivKey 'j'
+educatorKPrivKey = mkPrivKey 'k'
+
+educatorKKeyPair, studentAKeyPair :: (PublicKey, SecretKey)
+educatorKKeyPair = mkKeyPair 'k'
+studentAKeyPair = mkKeyPair 'a'
+
+mkPrivateTxAux :: PublicKey -- ^ Public key used to create student addr
+               -> (PublicKey, SecretKey) -- ^ Key pair used to create submission witness
+               -> (PublicKey, SecretKey) -- ^ Key pair used to create transaction witness
```

Comment:
As said in previous comment, we only need submission witness (provided by student). Also, in tx definitions you misplace student and educator keypairs, causing test to fail.